src/main/java/com/labsynch/labseer/service/SaltStructureServiceImpl.java

```diff
@@ -65,27 +65,106 @@ public Salt saveStructure(Salt salt) throws CmpdRegMolFormatException {
 
 	@Override
 	public Salt update(Salt salt) throws CmpdRegMolFormatException {
-		CmpdRegMolecule mol = chemStructureService.toMolecule(salt.getMolStructure());
-		salt.setOriginalStructure(salt.getMolStructure());
-		salt.setMolStructure(mol.getMolStructure());
-		salt.setFormula(mol.getFormula());
-		salt.setMolWeight(mol.getMass());
-		salt.setCharge(mol.getTotalCharge());
+		try
+		{
+			CmpdRegMolecule mol = chemStructureService.toMolecule(salt.getMolStructure());
+			salt.setOriginalStructure(salt.getMolStructure());
+			salt.setMolStructure(mol.getMolStructure());
+			salt.setFormula(mol.getFormula());
+			salt.setMolWeight(mol.getMass());
+			salt.setCharge(mol.getTotalCharge());
 
-		logger.debug("salt code: " + salt.getAbbrev());
-		logger.debug("salt name: " + salt.getName());
-		logger.debug("salt structure: " + salt.getMolStructure());
+			logger.debug("salt code: " + salt.getAbbrev());
+			logger.debug("salt name: " + salt.getName());
+			logger.debug("salt structure: " + salt.getMolStructure());
 
-		boolean updated = chemStructureService.updateStructure(mol, StructureType.SALT, salt.getCdId());
-		salt.merge();
+			boolean updated = chemStructureService.updateStructure(mol, StructureType.SALT, salt.getCdId());
+			salt.merge();
 
-		if (updated) {
-			return Salt.findSalt(salt.getId());
-		} else {
-			return null;
+			if (updated) {
+				return Salt.findSalt(salt.getId());
+			} else {
+				return null;
+			}
 		}
+		catch (Exception e)
+        {
+			e.printStackTrace();
+			return null;
+        }
+
 	}
 
+    public Salt edit(Salt oldSalt, Salt newSalt){
+        try
+        {
+            oldSalt.setAbbrev(newSalt.getAbbrev());
+            oldSalt.setName(newSalt.getName());
+
+            oldSalt.setOriginalStructure(newSalt.getMolStructure());
+        	oldSalt.setMolStructure(newSalt.getMolStructure());
+            oldSalt.setFormula(calculateFormula(newSalt));
+            oldSalt.setMolWeight(calculateWeight(newSalt));
+            oldSalt.setCharge(calculateCharge(newSalt));
+			
+			chemStructureService.updateStructure(oldSalt.getMolStructure(), StructureType.SALT, oldSalt.getCdId());
+
+            oldSalt.merge();
+			return oldSalt;
+        }
+        catch (Exception e)
+        {
+			e.printStackTrace();
+			return null;
+        }
+    }
+
+	public double calculateWeight(Salt salt) {
+        double weight = -1.0;
+        try
+        {
+            CmpdRegMolecule mol = chemStructureService.toMolecule(salt.getMolStructure());
+		    weight = mol.getExactMass();
+            return weight;
+        }
+        catch (Exception e)
```

Comment:
I'm not sure why these general "catch" statements are here. Is there a reason we don't simply let any errors get thrown and bubble up to whatever code is calling `calculateWeight`? I'm generally opposed to "swallowing" errors unless we have a specific reason for a specific type of error.

Reply:
Will remove them or make more specific catches